File: src/LearnMS.API/Entities/CreditCode.cs
namespace LearnMS.API.Entities;

public sealed class CreditCode
{
    public required string Code { get; set; }
    public decimal Value { get; set; }
    public Guid? AssistantId { get; set; }
    public Guid? StudentId { get; set; }
    public CreditCodeStatus Status { get; set; } = CreditCodeStatus.Fresh;

}

public enum CreditCodeStatus
{
    Fresh,
    Sold,
    Redeemed
}

File: src/LearnMS.React/src/types/credit-code.ts
export type CreditCode = {
  code: string;
  value: number;
  status: "Fresh" | "Sold" | "Redeemed";
  redeemer: {
    id: string;
    email: string;
  };
  generator: {
    id: string;
    email: string;
  };
};
